src/MatrixTwitter.js

```diff
@@ -0,0 +1,1040 @@
+ */
+MatrixTwitter.prototype.start = function () {
+  if(this.app_twitter != null) {
+    log.warn("Twitter",  "Attempted to call start() while having been started previously.");
+    return
+  }
+
+  return this._get_bearer_token().then((token) => {
+    this.app_auth.bearer_token = token;
+    log.info('Twitter', 'Retrieved token');
+    this.app_twitter = new Twitter(this.app_auth);
+    this.msg_queue_intervalID = setInterval(() => {
+      this._process_head_of_msg_queue();
+    }, TWITTER_MSG_QUEUE_INTERVAL_MS);
+
+    this.start_timeline();
+    this.start_hashtag();
+
+  }).catch((error) => {
+    log.error('Twitter', 'Error trying to retrieve bearer token:', error);
+    throw "Couldn't get a bearer token for Twitter AS.";
+  });
+}
+
+MatrixTwitter.prototype._get_bearer_http = function () {
+  return new Promise( (resolve, reject) => {
+    var key = this.app_auth.consumer_key + ":" + this.app_auth.consumer_secret;
+    key = Buffer.from(key, 'ascii').toString('base64');
+    var options = {
+      url: "https://api.twitter.com/oauth2/token",
+      headers: {
+        'Authorization': "Basic " + key
+      },
+      form: "grant_type=client_credentials",
+      contentType: "application/x-www-form-urlencoded;charset=UTF-8"
+    };
+    Request.post(options, function (error, response, body) {
+      if (error) {
+        reject(error);
+      } else if (response.statusCode !== 200) {
+        reject("Response to bearer token request returned non OK")
+        log.error("Twitter",
+              "Body of response:%s\nStatuscode of respnse:%s",
+              body,
+              response.statusCode
+            );
+      } else {
+        try {
+          var jsonresponse = JSON.parse(body);
+          if (jsonresponse.token_type == "bearer") {
+            fs.writeFile("bearer.tok", jsonresponse.access_token, (err) => {
+              if (err) {
+                            //This error is unfortunate, but not a failure to retrieve a token so the bridge can run fine.
+                log.error("Twitter", "Couldn't write bearer token to file. Reason:", err);
+              }
+            });
+                    //Not waiting for callback since it is trivial to get a new token, and can be done async
+            resolve(jsonresponse.bearer_token);
+          } else {
+            reject("Request to oauth2/post did not return the correct" +
+                    "token type ('bearer'). This is weeeird.");
+            log.error("Twitter", "Body of response:%s", body);
+          }
+        } catch (e) {
+          reject(e);
+        }
+      }
+    });
+  });
+}
+
+MatrixTwitter.prototype._get_bearer_token = function () {
+  return new Promise((resolve) => {
+    fs.readFile('bearer.tok', {encoding: 'utf-8'}, (err, content) => {
+      if(err) {
+        log.warn('Twitter', "Token file not found or unreadable. Requesting new token.");
+        log.error("Twitter", err);
+        resolve(Promise.resolve(this._get_bearer_http()));
```

Comment:
You don't need to wrap `this._get_bearer_http()` in `Promise.resolve` as it already returns a promise.